common/scala/src/main/scala/org/apache/openwhisk/http/PoolingRestClient.scala

```diff
@@ -110,9 +110,16 @@ class PoolingRestClient(
       if (response.status.isSuccess) {
         Unmarshal(response.entity.withoutSizeLimit).to[T].map(Right.apply)
       } else {
-        // This is important, as it drains the entity stream.
-        // Otherwise the connection stays open and the pool dries up.
-        response.discardEntityBytes().future.map(_ => Left(response.status))
+        Unmarshal(response.entity).to[String].flatMap { errorDetails =>
+          val statusCode = response.status
+          val statusCodeReason = statusCode.reason()
+          val customReason = s"$statusCodeReason (details:$errorDetails)"
```

Comment:
I'd call out that `details:` is actually the response's `body:`. Also please add a space char for good measure.

Reply:
@markusthoemmes, @rabbah thanks for the final comments!